general/g.version/main.c

```diff
@@ -140,26 +144,18 @@ int main(int argc, char *argv[])
     }
 
     if (gish_rev->answer) {
-	char **rev_ver = G_tokenize(GIS_H_VERSION, "$");
-	char **rev_time = G_tokenize(GIS_H_DATE, "$");
-	const int tokens_expected = 3;
+	char *rev_ver = GIS_H_VERSION;
+	char *rev_time = GIS_H_DATE;
 	int no_libgis = FALSE;
 
-	/* if number of tokes is right, print it */
-	if (G_number_of_tokens(rev_ver) == tokens_expected &&
-	    G_number_of_tokens(rev_time) == tokens_expected) {
+	if (*rev_ver && *rev_time) {
 	    if (shell->answer) {
-                const char *p;
-                p = strstr(rev_ver[1], " ");
-		fprintf(stdout, "libgis_revision=%s\n",
-			p ? p + 1 : "00000");
-                p = strstr(rev_time[1], " ");
-		fprintf(stdout, "libgis_date=\"%s\"\n",
-			p ? p + 1 : "?");
+		fprintf(stdout, "libgis_revision=%s\n", rev_ver);
```

Comment:
```c
fprintf(stdout, "libgis_revision=\"%s\"\n", rev_ver);
```
rev_ver has spaces.

Reply:
Makes sense.
While on it, generally shell script style prints to stdout, otherwise G_message() is used (printing to to stderr), but g.version always prints to stdout. Make g.version use G_message() if shell script style is not requested, making it conforming to general GRASS message printing rules?

Reply:
Semantically, expected output of g.version is text (I would expect to be able to grab it from stdout in this case, shell style or not), so I would say just leave it. g.region also prints to stdout even for non-shell styles.

I believe that G_message() should only be used for informational messages that are not part of actual text output.

Reply:
Right, expected output of g.version is text and should go to stdout, shell style or not.